### ISSUE TITLE
ChunkedEncodingError in QueryUniprot

### ISSUE THREAD

**Issue by saramsey** [RTXteam/RTX#143]
File "<timeit-src>", line 6, in inner
  File "BuildMasterKG.py", line 120, in <lambda>
    running_time = timeit.timeit(lambda: run_function(), number=1)
  File "BuildMasterKG.py", line 97, in make_master_kg
    bne.expand_all_nodes()
  File "/mnt/data/orangeboard/RTX/code/reasoningtool/kg-construction/BioNetExpander.py", line 534, in expand_all_nodes
    self.expand_node(node)
  File "/mnt/data/orangeboard/RTX/code/reasoningtool/kg-construction/BioNetExpander.py", line 522, in expand_node
    expand_method(self)
  File "/mnt/data/orangeboard/RTX/code/reasoningtool/kg-construction/BioNetExpander.py", line 151, in expand_metabolite
    uniprot_ids = QueryUniprot.map_enzyme_commission_id_to_uniprot_ids(ec_id)
  File "/mnt/data/orangeboard/RTX/code/reasoningtool/kg-construction/QueryUniprot.py", line 106, in map_enzyme_commission_id_to_uniprot_ids
    res = QueryUniprot.__access_api(handler)
  File "/mnt/data/orangeboard/RTX/code/reasoningtool/kg-construction/QueryUniprot.py", line 87, in __access_api
    res = requests.get(url, timeout=QueryUniprot.TIMEOUT_SEC, headers=header)
  File "/home/rt/.local/lib/python3.5/site-packages/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "/home/rt/.local/lib/python3.5/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/requests_cache/core.py", line 126, in request
    **kwargs
  File "/home/rt/.local/lib/python3.5/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.5/dist-packages/requests_cache/core.py", line 99, in send
    return send_request_and_cache_response()
  File "/usr/local/lib/python3.5/dist-packages/requests_cache/core.py", line 91, in send_request_and_cache_response
    response = super(CachedSession, self).send(request, **kwargs)
  File "/home/rt/.local/lib/python3.5/site-packages/requests/sessions.py", line 658, in send
    r.content
  File "/home/rt/.local/lib/python3.5/site-packages/requests/models.py", line 823, in content
    self._content = bytes().join(self.iter_content(CONTENT_CHUNK_SIZE)) or bytes()
  File "/home/rt/.local/lib/python3.5/site-packages/requests/models.py", line 748, in generate
    raise ChunkedEncodingError(e)
requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))

**Comment by saramsey** [comment]
This SO posting seems relevant:
https://stackoverflow.com/questions/44509423/python-requests-chunkedencodingerrore-requests-iter-lines

**Comment by DeqingQu** [comment]
I guess this error might be caused by out of time. When I tested the api, the reponse time will be quite long if it will return too many ids.

> This error is complaining about one of two things: either that it's trying to read a chunk length but one hasn't been sent, or that it's trying to read a length-1 chunk. My bet is on the first. I'm prepared to believe that if the read times out before a chunk length is received, we throw this exception, but it's also possible the connection is being dropped.

https://github.com/ryanmcgrath/twython/issues/288

**Comment by saramsey** [comment]
@DeqingQu thank you!  What are your recommendations?

**Comment by saramsey** [comment]
Also this error occurred once after the BuildMasterKG.py script was running for 1.5 days (it was halfway done building the master KG), so it is infrequent. If it is a *random* and infrequent error, we might be able to get a clean successful run of BuildMasterKG.py before this error occurs again. Based on that reasoning I have restarted the script on rtxsteve.saramsey.org, to see if the error happens again.

**Comment by DeqingQu** [comment]
@saramsey 
I wrote a mock backend to test this weird error. If the response time exceeds 120 seconds, my code will throw an exception and return an empty set as I expect. So the ChunkedEncodingError may not be related to timeout.
When I checked the code more carefully, I found it only handled the requests.exceptions.Timeout exception. It is very dangerous. If any other exception happens, like ChunkedEncodingError, it will be crushed. 
All the QueryXXXX modules have the same bug. Should I fix all of them now or only fix the QueryUniProt.py module now and fix others after Hackathon?
